components/match2/wikis/mobilelegends/match_group_input_custom.lua

```diff
@@ -7,523 +7,210 @@
-			data.winner = 0
-			data.resulttype = 'draw'
-			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, 'draw')
-		elseif CustomMatchGroupInput.placementCheckSpecialStatus(indexedScores) then
-			data.winner = MatchGroupInput.getDefaultWinner(indexedScores)
-			data.resulttype = DEFAULT_RESULT_TYPE
-			if MatchGroupInput.hasForfeit(indexedScores) then
-				data.walkover = 'ff'
-			elseif MatchGroupInput.hasDisqualified(indexedScores) then
-				data.walkover = 'dq'
-			elseif MatchGroupInput.hasDefaultWinLoss(indexedScores) then
-				data.walkover = 'l'
-			end
-			indexedScores = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, DEFAULT_RESULT_TYPE)
-		else
-			local winner
-			indexedScores, winner = CustomMatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
-			data.winner = data.winner or winner
-		end
-	end
+	match.games = games
+	match.opponents = opponents
 
-	--set it as finished if we have a winner
-	if not Logic.isEmpty(data.winner) then
-		data.finished = true
-	end
+	match.extradata = MatchFunctions.getExtraData(match)
 
-	return data, indexedScores
+	return match
 end
 
+---@param match table
 ---@param opponents table[]
----@param winner integer?
----@param specialType string?
----@param finished boolean|string?
 ---@return table[]
----@return integer?
-function CustomMatchGroupInput.setPlacement(opponents, winner, specialType, finished)
-	if specialType == 'draw' then
-		for key, _ in pairs(opponents) do
-			opponents[key].placement = 1
-		end
-	elseif specialType == DEFAULT_RESULT_TYPE then
-		for key, _ in pairs(opponents) do
-			if key == winner then
-				opponents[key].placement = 1
-			else
-				opponents[key].placement = 2
-			end
-		end
-	else
-		local lastScore = NO_SCORE
-		local lastPlacement = NO_SCORE
-		local counter = 0
-		for scoreIndex, opp in Table.iter.spairs(opponents, CustomMatchGroupInput.placementSortFunction) do
-			local score = tonumber(opp.score)
-			counter = counter + 1
-			if counter == 1 and (winner or '') == '' then
-				if finished then
-					winner = scoreIndex
-				end
-			end
-			if lastScore == score then
-				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement or '') or lastPlacement
-			else
-				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement or '') or counter
-				lastPlacement = counter
-				lastScore = score or NO_SCORE
-			end
+function CustomMatchGroupInput.extractMaps(match, opponents)
+	local maps = {}
+	for key, map, mapIndex in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.vod = map.vod or String.nilIfEmpty(match['vodgame' .. mapIndex])
+		map.participants = MapFunctions.getParticipants(map, opponents)
+		map.extradata = MapFunctions.getExtraData(map, #opponents)
+
+		local opponentInfo = Array.map(opponents, function(_, opponentIndex)
```

Comment:
```suggestion
		map.opponents = Array.map(opponents, function(_, opponentIndex)
```
would save a tiny bit of refactoring with follow ups of #4822

Reply:
Rather not, as `opponents` and `participants` props are not intended to be use together with current live implemenation

Reply:
alrighty